File: Codellama-34b-Instruct/Template_and_context_final/kernels_construct_self_clause.f90
fortran
#ifndef T1
!T1:kernels construct self clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER :: device_type
        LOGICAL :: self_clause_present
        
        !$acc kernels self(device_type == acc_device_host)
        !$acc end kernels
        
        IF (device_type == acc_device_host) THEN
          self_clause_present = .TRUE.
        ELSE
          self_clause_present = .FALSE.
        END IF
        
        IF (self_clause_present) THEN
          errors = errors + 1
        END IF
        
        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM

File: gpt-3.5-ft-large/template-context-full/kernels_construct_self_clause.cpp
#include "acc_testsuite.h"
#ifndef T1
//T1:kernels construct self clause,V:2.7-3.3
int test1(){
    int err = 0;
    srand(SEED);

    if(true){
        #pragma acc kernels self(conditional: _dev_a[0:n])
            for(i = 0; i < n; i++){
                _dev_a[i] = 1;
            }
    }
        
         if(_dev_a.lengh > 0){
            for(int x = 0; x < _dev_a.length; ++x){
                if(_dev_a[x] != 1.0){
                    err = 1;
                }
            }
        }
    

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}